### DISCUSSION TITLE
请问能不能既用class又用printf和scanf，会显得语言风格很混乱吗

### DISCUSSION THREAD

**Discussion by myalice-forever** [liusy58/NJU_AdvancedProgramming_SP22#16] [Q&A] [answered]
如上

**Comment by Price1999a** [comment]
众所周知 g++ msvc clang 都支持printf。

因此，不是问题。

**Comment by liusy58** [comment, marked as answer]
事实上，`printf`依然存在在很多的C++项目中，[Is it a bad practice to use printf in C++?](https://www.quora.com/Is-it-a-bad-practice-to-use-printf-in-C) 也许是你想问的，据我所知，JVM中存在大量的`printf`，还是一样的道理，C++会向下兼容很多的东西，虽然不能说C++是C的超集，但是确实很多东西C++都会保持兼容，如果你认为一些风格混乱，我认为这样的情况存在一定的风格混乱，比如你用了C++的`using`关键字，然后你还用在C语言中经常出现的`typedef`。

**Comment by RicoloveFeng** [comment]
<s>Try `std::format`.</s>

**Reply by cqjjjzr** [reply]
热知识：2022/02/23了 GCC 和 libstdc++ 还没有支持 `std::format`

**Reply by RicoloveFeng** [reply]
> 热知识：2022/02/23了 GCC 和 libstdc++ 还没有支持 `std::format`

🤨那这也太fw了吧，我以为2022年了，不应该还存在标准发布两年后编译器还没支持的情况……

**Comment by huanhuan6666** [comment]
printf的格式化输出还是很好用的，怎么方便怎么来，C++是C的超集，不能有了OO思想就忘了亲爹
